system-contracts/bootloader/bootloader.yul

```diff
@@ -2414,7 +2610,21 @@ object "Bootloader" {
                 setHook(VM_HOOK_ACCOUNT_VALIDATION_ENTERED())
                 debugLog("pre-validate",0)
                 debugLog("pre-validate",from)
+
+                // Override bytecode hash for validation if required.
+                // TODO: It should be safe, since delegation is only allowed for EOAs in the first place.
+                let delegation := getDelegationAddress(from)
+                let rawCodeHash := 0
+                if gt(delegation, 0) {
+                    rawCodeHash := getRawCodeHash(delegation, true)
```

Comment:
I have a somewhat bad feeling about this hack when it is coupled with a paymaster:
- Paymaster can do any calls
- But the world is in a strange state where the account actually has a delegator but the bytecodes are different

To make it safer one option could be similar to what I described in another thread: keep `DefaultAccount` code roughly same, but also store the delegate address.

Another alternative if we decide to stick to the bytecode-based approach is to disallow paymasters